airbyte-commons-worker/src/main/java/io/airbyte/workers/normalization/NormalizationRunnerFactory.java

```diff
@@ -38,12 +41,22 @@ public class NormalizationRunnerFactory {
 
   public static NormalizationRunner create(final String connectorImageName,
                                            final ProcessFactory processFactory,
-                                           final String normalizationVersion) {
+                                           final String normalizationVersion,
+                                           final String normalizationImage) {
     final var valuePair = getNormalizationInfoForConnector(connectorImageName);
+    final String factoryNormalizationImage = String.format("%s:%s", valuePair.getLeft(), normalizationVersion);
+    log.error("destination normalization runner factory -> {}", factoryNormalizationImage);
+    if (Objects.nonNull(normalizationImage)
+        && !normalizationImage.equalsIgnoreCase(factoryNormalizationImage)) {
+      log.error(
+          "The normalization image name or tag in the definition file is different from the normalization image or tag in the NormalizationRunnerFactory!");
+      log.error(
+          "the definition file value - {}, the NormalizationRunnerFactory value - {}", normalizationImage, factoryNormalizationImage);
+    }
```

Comment:
for my understanding, this is something that could happen in the meantime before removing NormalizationRunnerFactory because the definitions could have been updated since then, but we would still be using the normalization image from the definitions, right?

Is the NormalizationRunnerFactory or the normalization mappings being used for anything at this point or could we get rid of it as part of this PR?

Reply:
Yes, we can remove the **NormalizationRunnerFactory** in this pull request. Maybe I'm playing it safe, but I'm doing everything according to the documentation, and it says to do it in separate stages =) And yes, in the future we will use the values from the definition file. Before merging with the master, I will definitely check this point. But as far as I understand, before publishing each new version of the connector, this file is updated, thereby we update the **actor_definition** table. I'm right?